www/console_view/src/module/main.module.spec.js

```diff
@@ -0,0 +1,225 @@
+        expect(state.controllerAs).toBe('c');
+        expect(state.templateUrl).toBe(`console_view/views/${name}.html`);
+        return expect(state.url).toBe(`/${name}`);
+    });
+});
+
+describe('Console view controller', function() {
+    // Test data
+
+    let $rootScope, $timeout, $window, dataService, scope;
+    let builders = [{
+        builderid: 1,
+        masterids: [1]
+    }
+    , {
+        builderid: 2,
+        masterids: [1]
+    }
+    , {
+        builderid: 3,
+        masterids: [1]
+    }
+    , {
+        builderid: 4,
+        masterids: [1]
+    }
+    ];
+
+    const builds1 = [{
+        buildid: 1,
+        builderid: 1,
+        buildrequestid: 1
+    }
+    , {
+        buildid: 2,
+        builderid: 2,
+        buildrequestid: 1
+    }
+    , {
+        buildid: 3,
+        builderid: 4,
+        buildrequestid: 2
+    }
+    , {
+        buildid: 4,
+        builderid: 3,
+        buildrequestid: 2
+    }
+    ];
+
+    const builds2 = [{
+        buildid: 5,
+        builderid: 2,
+        buildrequestid: 3
+    }
+    ];
+
+    const builds = builds1.concat(builds2);
+
+    const buildrequests = [{
+        builderid: 1,
+        buildrequestid: 1,
+        buildsetid: 1
+    }
+    , {
+        builderid: 1,
+        buildrequestid: 2,
+        buildsetid: 1
+    }
+    , {
+        builderid: 1,
+        buildrequestid: 3,
+        buildsetid: 2
+    }
+    ];
+
+    const buildsets = [{
+        bsid: 1,
+        sourcestamps: [
+            {ssid: 1}
+        ]
+    }
+    , {
+        bsid: 2,
+        sourcestamps: [
+            {ssid: 2}
+        ]
+    }
+    ];
+
+    const changes = [{
+        changeid: 1,
+        sourcestamp: {
+            ssid: 1
+        }
+    }
+    ];
+    let createController = (scope = ($rootScope = (dataService = ($window = ($timeout = null)))));
+
+    const injected = function($injector) {
+        const $q = $injector.get('$q');
+        $rootScope = $injector.get('$rootScope');
+        $window = $injector.get('$window');
+        $timeout = $injector.get('$timeout');
+        dataService = $injector.get('dataService');
+        scope = $rootScope.$new();
+        dataService.when('builds', builds);
+        dataService.when('builders', builders);
+        dataService.when('changes', changes);
+        dataService.when('buildrequests', buildrequests);
+        dataService.when('buildsets', buildsets);
+
+        // Create new controller using controller as syntax
+        const $controller = $injector.get('$controller');
+        return createController = () =>
+            $controller('consoleController as c', {
+                // Inject controller dependencies
+                $q,
+                $window,
+                $scope: scope
+            }
+            )
+        ;
+    };
+
+    beforeEach(inject(injected));
+
+    it('should be defined', function() {
+        createController();
+        return expect(scope.c.crazy).toBeDefined();
```

Comment:
There is still a bunch of unneeded returns we need to get rid of them

Reply:
Thanks, yes it does seem like the tests in particular end up with unnecessary returns in them.  Since coffeescript has this property of returning without an explicit ```return``` statement, I guess the decaffeinate errs on the side of caution, re-adding ```return``` in many places where it's not needed.  Ill take a look for those.

So far I haven't been able to get the test code running.    Currently the package.json for this plugin doesn't include Angular as a dependency,  and this causes the tests to fail to run.  The plugin runs fine though because Angular is already loaded by the base www stuff before the plugin ever gets loaded.  I'm not sure what the correct approach is to properly add the dependency on Angular but without bundling a duplicate copy of Angular into the webpack bundle.